src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -226,48 +226,50 @@ public void missingSoundMapping(RegistryEvent.MissingMappings<SoundEvent> event)
-
-        event.getRegistry().register(new ConditionalRecipe.Serializer<Recipe<?>>().setRegistryName(new ResourceLocation("forge", "conditional")));
-
+        if (event.getRegistryKey().equals(ForgeRegistries.Keys.RECIPE_SERIALIZERS))
+        {
+            CraftingHelper.register(AndCondition.Serializer.INSTANCE);
+            CraftingHelper.register(FalseCondition.Serializer.INSTANCE);
+            CraftingHelper.register(ItemExistsCondition.Serializer.INSTANCE);
+            CraftingHelper.register(ModLoadedCondition.Serializer.INSTANCE);
+            CraftingHelper.register(NotCondition.Serializer.INSTANCE);
+            CraftingHelper.register(OrCondition.Serializer.INSTANCE);
+            CraftingHelper.register(TrueCondition.Serializer.INSTANCE);
+            CraftingHelper.register(TagEmptyCondition.Serializer.INSTANCE);
+
+            CraftingHelper.register(new ResourceLocation("forge", "compound"), CompoundIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "nbt"), NBTIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "partial_nbt"), PartialNBTIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "difference"), DifferenceIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "intersection"), IntersectionIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("minecraft", "item"), VanillaIngredientSerializer.INSTANCE);
+
+            event.register(ForgeRegistries.Keys.RECIPE_SERIALIZERS, new ResourceLocation("forge", "conditional"), new ConditionalRecipe.Serializer<>());
+        }
     }
 
-    @SubscribeEvent //ModBus
-    @SuppressWarnings("unused") // automatically called by Forge
-    public void registerLootData(RegistryEvent.Register<GlobalLootModifierSerializer<?>> event)
+    public void registerLootData(RegisterEvent event)
     {
-        // Ignore the event itself: this is done only not to statically initialize our custom LootConditionType
-        Registry.register(Registry.LOOT_CONDITION_TYPE, new ResourceLocation("forge:loot_table_id"), LootTableIdCondition.LOOT_TABLE_ID);
-        Registry.register(Registry.LOOT_CONDITION_TYPE, new ResourceLocation("forge:can_tool_perform_action"), CanToolPerformAction.LOOT_CONDITION_TYPE);
+        if (event.getRegistryKey().equals(ForgeRegistries.Keys.LOOT_MODIFIER_SERIALIZERS))
+        {
+            // Ignore the event itself: this is done only not to statically initialize our custom LootConditionType
+            Registry.register(Registry.LOOT_CONDITION_TYPE, new ResourceLocation("forge:loot_table_id"), LootTableIdCondition.LOOT_TABLE_ID);
+            Registry.register(Registry.LOOT_CONDITION_TYPE, new ResourceLocation("forge:can_tool_perform_action"), CanToolPerformAction.LOOT_CONDITION_TYPE);
+        }
```

Comment:
Looks like this existed as a case of registering objects to vanilla registries in the `RegistryEvent.Register` of the closest type that is related to what's being registered. This should be modified to listen for the `LOOT_CONDITION_TYPE` event and register using the event's `register` method.